core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1529,6 +1550,18 @@ public void run() {
 
     @Override
     public void requestConnection() {
+      syncContext.execute(new Runnable() {
```

Comment:
Why was this changed? It seems it should be safe to check `started` here without going onto the syncContext.

Reply:
Reverted per our discussion. There is a happens-after relationship between `start()` and the other methods, thus `started` don't need to be read in the syncContext.